javav2/example_code/s3/src/main/java/com/example/s3/batch/S3BatchActions.java

```diff
@@ -0,0 +1,705 @@
+                System.err.println("Failed to cancel job: " + ex.getMessage());
+                throw new RuntimeException(ex); // Propagate the exception
+            });
+    }
+    // snippet-end:[s3control.java2.cancel_job.main]
+
+    // snippet-start:[s3control.java2.update_job.main]
+    /**
+     * Updates the priority of a job asynchronously.
+     *
+     * @param jobId     the ID of the job to update
+     * @param accountId the ID of the account associated with the job
+     * @return a {@link CompletableFuture} that represents the asynchronous operation, which completes when the job priority has been updated or an error has occurred
+     */
+    public CompletableFuture<Void> updateJobPriorityAsync(String jobId, String accountId) {
+        UpdateJobPriorityRequest priorityRequest = UpdateJobPriorityRequest.builder()
+            .accountId(accountId)
+            .jobId(jobId)
+            .priority(60)
+            .build();
+
+        CompletableFuture<Void> future = new CompletableFuture<>();
+        getAsyncClient().updateJobPriority(priorityRequest)
+            .thenAccept(response -> {
+                System.out.println("The job priority was updated");
+                future.complete(null); // Complete the CompletableFuture on successful execution
+            })
+            .exceptionally(ex -> {
+                System.err.println("Failed to update job priority: " + ex.getMessage());
+                future.completeExceptionally(ex); // Complete the CompletableFuture exceptionally on error
+                return null; // Return null to handle the exception
+            });
+
+        return future;
+    }
+    // snippet-end:[s3control.java2.update_job.main]
+
+    // snippet-start:[s3control.java2.list_jobs.main]
+    /**
+     * Asynchronously lists batch jobs that have completed for the specified account.
+     *
+     * @param accountId the ID of the account to list jobs for
+     * @return a CompletableFuture that completes when the job listing operation is finished
+     */
+    public CompletableFuture<Void> listBatchJobsAsync(String accountId) {
+        ListJobsRequest jobsRequest = ListJobsRequest.builder()
+            .jobStatuses(JobStatus.COMPLETE)
+            .accountId(accountId)
+            .maxResults(10)
+            .build();
+
+        return asyncClient.listJobs(jobsRequest)
+            .thenAccept(response -> {
+                List<JobListDescriptor> jobs = response.jobs();
+                for (JobListDescriptor job : jobs) {
+                    System.out.println("The job id is " + job.jobId());
+                    System.out.println("The job priority is " + job.priority());
+                }
+            })
+            .exceptionally(ex -> {
+                System.err.println("Failed to list batch jobs: " + ex.getMessage());
+                throw new RuntimeException(ex); // Propagate the exception
+            });
+    }
+    // snippet-end:[s3control.java2.list_jobs.main]
```

Comment:
Pagination is not implemented here, and it's necessary for listing all jobs.
Add comments to describe the intent and use of pagination.